src/base/bittorrent/infohash.h

```diff
@@ -28,41 +28,25 @@
 
 #pragma once
 
-#include <libtorrent/sha1_hash.hpp>
-
+#include <QHash>
 #include <QMetaType>
-#include <QString>
+
+#include "base/digest32.h"
+
+using SHA1Hash = Digest32<160>;
```

Comment:
would it make sense to put them under `BitTorrent` namespace?

Reply:
>would it make sense to put them under `BitTorrent` namespace?

No. `InfoHash` uses `SHA1Hash` (and not abstract `Digest32<160>`) as base class so it should be declared before.